diff --git a/var/lib/lxc/android/pre-start.sh b/var/lib/lxc/android/pre-start.sh
index 8006486..999b75e 100755
--- a/var/lib/lxc/android/pre-start.sh
+++ b/var/lib/lxc/android/pre-start.sh
@@ -16,12 +16,6 @@ mount_android_partitions() {
 	done
 }
 
-if [ -f "/tmp/lxc_android_once" ]; then
-    echo "lxc:android contianer had already been invoked.";
-    exit -255
-fi
-touch /tmp/lxc_android_once
-
 INITRD=/system/boot/android-ramdisk.img
 rm -Rf $LXC_ROOTFS_PATH
 mkdir -p $LXC_ROOTFS_PATH
