From 66d81409d0ae8bcd25e0226f9bdfec24de44b76b Mon Sep 17 00:00:00 2001
From: Muhammad <muhammad23012009@hotmail.com>
Date: Sun, 9 Apr 2023 11:57:49 +0500
Subject: [PATCH] Makefile: install to /usr/include/android-28

Signed-off-by: Muhammad <muhammad23012009@hotmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2403747..62d431b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 PREFIX?=/usr/local
-INCLUDEDIR?=$(PREFIX)/include/android
+INCLUDEDIR?=$(PREFIX)/include/android-28
 PKGCONFIGDIR?=$(PREFIX)/lib/pkgconfig
 all:
 	@echo "Use '$(MAKE) install' to install"
-- 
2.39.2

