From 0ed2f872fa6dce04199edda0d1e600c341cdd4fe Mon Sep 17 00:00:00 2001
From: Muhammad <muhammad23012009@hotmail.com>
Date: Sun, 9 Apr 2023 11:56:31 +0500
Subject: [PATCH] makefile: install to /usr/include/android-29

Signed-off-by: Muhammad <muhammad23012009@hotmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6c15a15..d7a48f6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 PREFIX?=/usr/local
-INCLUDEDIR?=$(PREFIX)/include/android
+INCLUDEDIR?=$(PREFIX)/include/android-29
 PKGCONFIGDIR?=$(PREFIX)/lib/pkgconfig
 all:
 	@echo "Use '$(MAKE) install' to install"
-- 
2.39.2

