From: Bardia Moshiri <fakeshell@bardia.tech>
diff -Npur a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -20,7 +20,7 @@
                                 'gtk_doc=false',
                                 'tests=false',
                               ])
-  gmobile        = gmobile_subp.get_variable('libgmobile_dep')
+  gmobile        = gmobile_subp.get_variable('gmobile_dep')
   gsound         = dependency('gsound')
   gudev          = dependency('gudev-1.0', version: '>=232')
   json_glib      = dependency('json-glib-1.0')
